replayer/neard/neard.go

```diff
@@ -29,28 +61,70 @@ func Build(release bool) error {
 	if release {
 		args = append(args, "--release")
 	}
+
 	cmd := exec.Command("cargo", args...)
+	cmd.Dir = repoPath
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
 	return cmd.Run()
 }
 
-func initDaemon(release bool, localDir string) error {
-	var name string
+// LoadFromRepo loads NEARDaemon from repo by provided path (and build if requested).
+func LoadFromRepo(repoPath string, release bool, build bool) (*NEARDaemon, error) {
+	log.Info("load neard repo")
+
+	head, err := git.Head(repoPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if build {
+		if err := buildBinary(repoPath, release); err != nil {
+			return nil, err
+		}
+	}
+
+	binaryPath := filepath.Join(repoPath, "target", "debug", "neard")
 	if release {
-		name = filepath.Join(".", "target", "release", "neard")
+		binaryPath = filepath.Join(repoPath, "target", "release", "neard")
+	}
+
+	return LoadFromBinary(binaryPath, head)
+}
+
+// Backup local data if it exists
+func (daemon *NEARDaemon) backupLocalData() error {
+	exists, err := file.Exists(daemon.localDir)
+	if err != nil {
+		return err
+	}
+	if exists {
+		localOld := strings.TrimSuffix(daemon.localDir, "/") + "_old"
```

Comment:
It's good style to use functions "path/filepath" for path manipulation and not assume "/" as a path separator, for example.

It's unlikely we are going to run this on Windows, but nonetheless a good habit ;)